docs/chapters/architecture/DLT/diagnostics-log-and-tracing.rst

```diff
@@ -0,0 +1,181 @@
+
+On a PELUX image, the dlt-daemon package is already integrated and enabled.
+
+In order to get DLT logs remotely, `dlt-viewer`_ is required.
+
+Basic usage
+"""""""""""
+The dlt-daemon runs by default when the PELUX image boots up. dlt-daemon usage can be found on https://github.com/GENIVI/dlt-daemon.  Cheat-sheets and further documentation can be found on https://github.com/GENIVI/dlt-daemon/tree/master/doc (Although the documentation here is outdated, it can be very helpful giving insight into architecture).
+
+dlt-daemon stores offline-log database  on /tmp file system.
+
+In order to view the logs via dlt-viewer, you need to install it on your machine first. Instructions are available on `the repository`_. Once dlt-viewer is built, its binaries are found on build/bin/dlt-viewer and the GUI can be launched via command ./dlt-viewer .  You will need to find the IP address of the host, finding the IP address depends on the hardware where PELUX is running.
+
+Basic requirements
+""""""""""""""""""
+**Unicode support**
+PELUX default encoding is UTF-8 for all logging messages. GENIVI dlt-daemon works well with UTF-8 as well.
+
+**Dynamic Log adjustments**
+The amount of logs, especially in Trace mode, can be overwhelming. Applications that produce DLT logs should have the functionality of enabling/disabling logs or adjusting the level of log outputs. Ideally each logging call should dynamically check boolean condition and just do nothing if logs are dynamically disabled. It will reduce log traffic inside the system and improve performance in general.
+
+GENIVI DLT provides a back channel from dlt-viewer to applications that generate DLT logs. The user can precisely change log levels for each application individually from the dlt-viewer UI. The user request will reach each application and disable logs in the place they are actually produced.
+
+Quality Goals
+"""""""""""""
+*Reliability*
+Reliability is a key quality for DLT. Developers and testers rely on the logs provided by dlt-daemon to find errors and understand behavior of applications. If the logs will be faulty, there will be a very big struggle to debug the error, or understand the behavior.
+
+Stakeholders
+^^^^^^^^^^^^
+* Software Developers
+* QA Engineers
+
+Constraints
+-----------
+GENIVI's dlt-daemon stores an offline log database on in-RAM file system /tmp/dlt_offlinetrace_20120101_005302.dlt . The size limits for the log files should be adjusted at HU startup based on amount of RAM available. E.g. a Premium HU with 16GB of RAM can use 200MB of offline storage. On the other hand, 4GB of ram uses 50MB of offline storage or less.
+
+Context
+-------
+Business Context
+^^^^^^^^^^^^^^^^
+DLT logs are automotive industry standard. Industry expects any automotive software to support it.
+
+Solution Strategy
+-----------------
+
+Since PELUX is a part of the open source community, we use a component developed by GENIVI.
```

Comment:
This and the "Business context" is what I feel I am missing in the introduction. Some mention of choosing GENIVI since it is OSS and developed for the Automotive industry of something.